examples/cmake/find_package_static/CMakeLists.txt

```diff
@@ -42,3 +42,4 @@ include_directories (${MONGOC_INCLUDE_DIRS})
 add_executable (hello_mongoc ../../hello_mongoc.c)
 target_include_directories (hello_mongoc PRIVATE "${MONGOC_INCLUDE_DIRS}")
 target_link_libraries (hello_mongoc PRIVATE "${MONGOC_LIBRARIES}")
+target_compile_definitions (hello_mongoc PRIVATE "${MONGOC_DEFINITIONS}")
```

Comment:
Let's do this in examples/cmake/find_package/CMakeLists.txt, too, to encourage users to write forward-compatible CMake files. Someday MONGOC_DEFINITIONS may be nonempty even for targets that link dynamically to libmongoc.